File: tape_equilibrium/tape_equilibrium.rb
def solution(a)
  min_diff = 2000
  length = a.length
  left_sum = a.shift
  right_sum = a.inject(0, :+)
  
  i=0
  while i < length-1
    diff = (left_sum - right_sum).abs
    if diff < min_diff 
      min_diff = diff
    end
    left_sum += a[i]
    right_sum -= a[i]
    i += 1
  end
  return min_diff
end

File: tape_equilibrium/tape_equilibrium.js
function solution(A) {
    var left_sum = A.shift();
    var right_sum = A.reduce(function(a,b) {
        return a+b    
    });
    var min_diff = Math.abs(left_sum - right_sum);
    for(var i = 0; i < A.length; i++) {
        var current_diff = Math.abs(left_sum - right_sum);
        if(current_diff < min_diff) {
            min_diff = current_diff    
        };
        left_sum += A[i];
        right_sum -= A[i];
    }
    return min_diff
}